### PR TITLE
Pull request for merging Introduction to Ruby on Rails

### PR DESCRIPTION
**Pull request by selina-sha**
This PR include the markdown file for Introduction to Ruby on Rails. It includes an introduction to what Rails is and a series of steps to quickly set up a rails application for beginners.

### CODE REVIEW

**Comment by wezhubb** [comment]
Nicely written page with lots of info! Great work!
Some areas of improvement could be:
I noticed there is a typo in the installation step 3. It should be “gem install” not “gen install”
I know that rails,  it has two major principles. Do you think it would be beneficial for beginners to know them?
Because Rails is built on Ruby the programming language. For beginners who don’t necessarily know ruby, it would be great if you could include some resources for learning ruby.

**Comment by selina-sha** [comment]
> Nicely written page with lots of info! Great work! Some areas of improvement could be: I noticed there is a typo in the installation step 3. It should be “gem install” not “gen install” 

Good catch!! Thank you so much! I will correct the typo.

**Comment by selina-sha** [comment]
> I know that rails, it has two major principles. Do you think it would be beneficial for beginners to know them? 

You are absolutely correct on this one! I think the two major principles you mentioned are "Don't Repeat Yourself" and "Convention Over Configuration". These are very important rules to keep in mind for developers, otherwise they would have a hard time understand why Rails is opinionated, especially for beginners. I will add them in my next commit.

**Comment by selina-sha** [comment]
> Because Rails is built on Ruby the programming language. For beginners who don’t necessarily know ruby, it would be great if you could include some resources for learning ruby.

I agree. Thanks for bringing this up! I will add it in my next commit!

**Comment by selina-sha** [comment]
Hi @wezhubb, I have addressed all your comments. Could you please review it again? Please let me know if you have any other suggestions! Thanks.

File: Topics/Tech_Stacks/Ruby_on_Rails.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+## Prerequisite
+This guide assumes that you have prior experience with Ruby, a programming language. You might find a steep learning curve if you dive straight into Rails.
+
+Here are some useful resources to learn Ruby:
+- [Official Ruby Website](https://www.ruby-lang.org/en/documentation/)
+- [Ruby Tutorial For Beginners](https://www.rubyguides.com/ruby-tutorial/)
+
+## What is Rails?
+Rails is a framework for web application development in the Ruby programming language. Its relationship to Ruby is similar to the one between Django and Python. The benefit of leveraging Rails is that it allows developers to write less code while still achieving more compared to other languages and frameworks. What makes Rails different from other frameworks is that it is opinionated software, designed with particular rules and preferences on how developers should write code.
```
Comment (wezhubb) [inline Topics/Tech_Stacks/Ruby_on_Rails.md:14]: Maybe explain further of what is Rail and what Rail could accomplish could help.

File: Topics/Tech_Stacks/Ruby_on_Rails.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+This guide assumes that you have prior experience with Ruby, a programming language. You might find a steep learning curve if you dive straight into Rails.
+
+Here are some useful resources to learn Ruby:
+- [Official Ruby Website](https://www.ruby-lang.org/en/documentation/)
+- [Ruby Tutorial For Beginners](https://www.rubyguides.com/ruby-tutorial/)
+
+## What is Rails?
+Rails is a framework for web application development in the Ruby programming language. Its relationship to Ruby is similar to the one between Django and Python. The benefit of leveraging Rails is that it allows developers to write less code while still achieving more compared to other languages and frameworks. What makes Rails different from other frameworks is that it is opinionated software, designed with particular rules and preferences on how developers should write code. 
+
+## Two Major Principles in Rails
```
Comment (wezhubb) [inline Topics/Tech_Stacks/Ruby_on_Rails.md:16]: Explain further why are those two principles important.

File: Topics/Tech_Stacks/Ruby_on_Rails.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+To start the web server, enter this command:
+```bash
+$ bin/rails server
+```
+Now, open your browser and enter `http://localhost:3000`. You should see the default Rails page. Congratulations! You have started your first Rails Applicaiton successfully!
+
+## Resources
+For more information on Rails, here are some useful websites:
+- [Rails Guides](https://guides.rubyonrails.org/getting_started.html). This guide covers all the topics mentioned in my article but at a more granular level.
+- [GoRails](https://gorails.com/start)
```
Comment (wezhubb) [inline Topics/Tech_Stacks/Ruby_on_Rails.md:86]: Overall, you should elaborate more on the reason for using Rail and different task Rail could achieve, current information only touches on the initial setup of Rail.

**Review by wezhubb — CHANGES_REQUESTED** [review]
Please review the comments, and let me know if there is any question.

File: Topics/Tech_Stacks/Ruby_on_Rails.md (hunk 1 above)
Comment (liuqin26) [reply Topics/Tech_Stacks/Ruby_on_Rails.md:14]: Can you also explain/rephrase "it allows developers to write less code while still achieving more compared to other languages and frameworks."? This sentence looks a bit ambiguous to me.

**Review by liuqin26 — COMMENTED** [review]
Hi Kexin,

I think you have excellent content on this page. However, I would suggest including a use case where a developer utilizes rails in Ruby to accomplish a task faster, so readers have an idea of when they are likely to find themselves needing rails. And lastly, I would recommend just going a little bit further on what's expected after the setup. I know this is only an introduction, but I am expecting just a taste of its functionality as a reader who is considering learning this. Great job overall!

Qingyi

File: Topics/Tech_Stacks/Ruby_on_Rails.md (hunk 1 above)
Comment (selina-sha) [reply Topics/Tech_Stacks/Ruby_on_Rails.md:14]: Thank you both for the comments! They are very helpful and I agree! I will address them in my next commit!

File: Topics/Tech_Stacks/Ruby_on_Rails.md (hunk 3 above)
Comment (selina-sha) [reply Topics/Tech_Stacks/Ruby_on_Rails.md:86]: Thanks for the feedback. I will expand more in my next commit!

**Comment by selina-sha** [comment]
Hi Team,

Thanks for the valuable feedbacks. I have addressed all the comments. Could you please review again? Thanks! @wezhubb @liuqin26

**Review by wezhubb — APPROVED** [review]

**Comment by wezhubb** [comment]
> Hi Team,
> 
> Thanks for the valuable feedbacks. I have addressed all the comments. Could you please review again? Thanks! @wezhubb @liuqin26

Hi @selina-sha,
I have reviewed and approved your changes. Nice work!

**Comment by selina-sha** [comment]
> I just reviewed your latest commit. Great job! I found the data model part particularly interesting and improved my understanding of what Rails does. I agree that we should merge this into main!

Thanks!!

**Pull request merged**